neuralop/layers/discrete_continuous_convolution.py

```diff
@@ -0,0 +1,611 @@
+    if periodic:
+        periodic_diffs = torch.where(diffs > 0.0, diffs - 1, diffs + 1)
+        diffs = torch.where(diffs.abs() < periodic_diffs.abs(), diffs, periodic_diffs)
+
+    r = torch.sqrt(diffs[0] ** 2 + diffs[1] ** 2)
+    phi = torch.arctan2(diffs[1], diffs[0]) + torch.pi
+
+    idx, vals = kernel_handle(r, phi)
+
+    idx = idx.permute(1, 0)
+
+    if normalize:
+        vals = _normalize_convolution_tensor_2d(idx, vals, grid_in, grid_out, kernel_shape, quad_weights, transpose_normalization=transpose_normalization)
+
+    return idx, vals
+
+
+class DiscreteContinuousConv2d(DiscreteContinuousConv):
+    """
+    Discrete-continuous convolutions (DISCO) on arbitrary 2d grids as implemented for [1].
+
+    [1] Liu-Schiaffini M., Berner J., Bonev B., Kurth T., Azizzadenesheli K., Anandkumar A.; Neural Operators with Localized Integral and Differential Kernels;  arxiv:2402.16845
+    [2] Ocampo J., Price M.A. , McEwen J.D.; Scalable and equivariant spherical CNNs by discrete-continuous (DISCO) convolutions, ICLR (2023), arXiv:2209.13603
+
+    Parameters
+        ----------
+        in_channels: int
+            input channels to DISCO convolution
+        out_channels: int
+            output channels of DISCO convolution
+        grid_in: torch.Tensor or str
+            input grid in the form of a point cloud. Can also pass a string to generate a regular (tensor) grid.
+        grid_out: torch.Tensor or str
+            output grid in the form of a point cloud. Can also pass a string to generate a regular (tensor) grid.
+        kernel_shape: Union[int, List[int]]
+            kernel shape. Expects either a signle integer for isotropic kernels or two integers for anisotropic kernels
+        n_in: Tuple[int], optional
+            number of input points
+        n_out: Tuple[int], optional
+            number of output points
+        quad_weights: torch.Tensor, optional
+            quadrature weights on the input grid
+        periodic: bool, optional
+            whether the domain is periodic
+        groups: int, optional
+            number of groups in the convolution
+        bias: bool, optional
+            whether to use a bias
+        radius_cutoff: float, optional
+            cutoff radius for the kernel
+
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        grid_in: torch.Tensor,
+        grid_out: torch.Tensor,
+        kernel_shape: Union[int, List[int]],
+        n_in: Optional[Tuple[int]] = None,
+        n_out: Optional[Tuple[int]] = None,
+        quad_weights: Optional[torch.Tensor] = None,
+        periodic: Optional[bool] = False,
+        groups: Optional[int] = 1,
+        bias: Optional[bool] = True,
+        radius_cutoff: Optional[float] = None,
+    ):
+        super().__init__(in_channels, out_channels, kernel_shape, groups, bias)
+
+        # the instantiator supports convenience constructors for the input and output grids
```

Comment:
This wording could be a little clearer - what does this mean?

Reply:
which wording exactly? Do you mean the variables in the `super().__init__` call?

Reply:
I mean, what is the context for 
```
# the instantiator....
```

This could be clearer